src/main/java/org/apache/maven/plugins/gpg/AbstractGpgMojo.java

```diff
@@ -138,23 +144,22 @@ public abstract class AbstractGpgMojo extends AbstractMojo {
     private String keyname;
 
     /**
-     * GPG Signer only: Passes <code>--use-agent</code> or <code>--no-use-agent</code> to gpg. If using an agent, the
-     * passphrase is optional as the agent will provide it. For gpg2, specify true as --no-use-agent was removed in
-     * gpg2 and doesn't ask for a passphrase anymore. Deprecated, and better to rely on session "interactive" setting
-     * (if interactive, agent will be used, otherwise not).
-     *
-     * @deprecated
+     * All signers: whether gpg-agent is allowed to be used or not. If enabled, passphrase is optional, as agent may
+     * provide it. Have to be noted, that in "batch" mode, gpg-agent will be prevented to pop up pinentry
+     * dialogue, hence best is to "prime" the agent caches beforehand.
+     * <p>
+     * GPG Signer: Passes <code>--use-agent</code> or <code>--no-use-agent</code> option to gpg if it is version 2.1
+     * or older. Otherwise, will use an agent. In non-interactive mode gpg options are appended with
+     * <code>--pinentry-mode error</code>, preventing gpg agent to pop up pinentry dialogue. Agent will be able to
+     * hand over only cached passwords.
+     * <p>
+     * BC Signer: Allows signer to communicate with gpg agent. In non-interactive mode it uses
+     * <code>--no-ask</code> option with the <code>GET_PASSPHRASE</code> function. Agent will be able to hand over
+     * only cached passwords.
      */
-    @Deprecated
     @Parameter(property = "gpg.useagent", defaultValue = "true")
     private boolean useAgent;
 
-    /**
-     * Detect is session interactive or not.
-     */
-    @Parameter(defaultValue = "${settings.interactiveMode}", readonly = true)
-    private boolean interactive;
```

Comment:
Why removing this parameter which provides a way to override the settings ?

Reply:
Ok, written like that, that clearly looks silly. I forgot that the settings was the one holding this bit, I thought it was on session.